### ISSUE TITLE
epoch1 cuda does not compile on the HTCondor GPU nodes (CentOS7)

### ISSUE THREAD

**Issue by lfield** [madgraph5/madgraph4gpu#92]
The epoch1 cuda code no longer compiles on the CentOS7 HTCondor nodes. It generates the following error.
```
nvcc warning : The -std=c++14 flag is not supported with the configured host compiler. Flag will be ignored.
In file included from /usr/include/c++/4.8.2/chrono:35:0,
                 from timer.h:4,
                 from timermap.h:14,
                 from gcheck.cu:25:
/usr/include/c++/4.8.2/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^
make: *** [gcheck.o] Error 1
```
After changing --std=c++14 to  --std=c++11, it generates:
```
timermap.h(111): error: namespace "std" has no member "defaultfloat"
timermap.h(132): error: namespace "std" has no member "defaultfloat"
gcheck.cu(440): error: namespace "std" has no member "defaultfloat"
gcheck.cu(640): error: namespace "std" has no member "defaultfloat"
gcheck.cu(654): error: namespace "std" has no member "defaultfloat"
5 errors detected in the compilation of "/pool/condor/dir_26045/tmpxft_00007236_00000000-6_gcheck.cpp1.ii".
```

**Comment by roiser** [comment]
Thanks for noticing, as cuda 10 supports gcc8 we had started to use this combination, but think it was never written down / nor checked somewhere. Could you pls upgrade your gcc version accordingly?

**Comment by lfield** [comment]
By default CentOS7 nodes have gcc 4.8.5. I was not aware that gcc8/CentOS8 was now a requirement but focusing on the latest and greatest makes sense. Is this an '_official_' decision?

**Comment by roiser** [comment]
This is the cuda/gcc version we used so far. As we are moving to cuda 11.0 we can also upgrade to gcc9. We should discuss this in the next meeting. 

For now I suggest gcc8 to be on the safe side.

**Comment by valassi** [comment]
Thanks Laurence. Actually I have even moved some of my code to require c++17 for vector extensions. I think gcc8 (the default on centos8) is the bare minimum. Anyway I am using that and it works, I see no reason to move to gcc9 yet. But gcc4.8 is way too old. Laurence you can always do simply
```
source /cvmfs/sft.cern.ch/lcg/releases/gcc/8.3.0/x86_64-centos7/setup.sh
```

**Comment by lfield** [comment]
This issue can be closed as it is now superseded by #96
